classes/services/PKPSchemaService.php

```diff
@@ -376,8 +378,8 @@ public function coerce($value, $type, $schema)
                 return $newArray;
             case 'object':
                 $newObject = []; // we handle JSON objects as assoc arrays in PHP
-                
-                if (isValidJson($value)) {
+
+                if (!is_object($value) && !is_array($value) && isValidJson($value)) {
```

Comment:
Hmm , since PHP 8+ , it got much stricter with type checking . The point is that `json_decode` now require first param to be `string` or throw fatal error while before PHP 8, it only throw a warning . How about rather than putting the check here , we put the check in the `isValidJson` method as such 
```php
if (!empty($data) && is_string($data)) { 
```
That way other also don't have be concern with these checks before using the `isValidJson` .

Reply:
I've updated the code and will go ahead and merge this